modules/dns_common/dns_common.rb

```diff
@@ -17,9 +19,28 @@ def initialize(server = nil, ttl = nil)
     end
 
     def resolver
+      logger.warn("Proxy::Dns::Record#resolver has been deprecated and will be removed, use get_dns_resource(s)")
       Resolv::DNS.new(:nameserver => @server)
     end
 
+    def get_dns_resource(name, typeclass)
+      result = nil
+      duration = time_spent_in_ms do
+        result = Resolv::DNS.new(:nameserver => @server).getresource(name, typeclass)
+      end
+      logger.info "Resolved #{name} via system resolver in #{duration.round(2)} ms"
```

Comment:
It's actually also incorrect. It uses specific nameservers from the configuration so it's not using the system resolver.

I'm against such verbose timing logging. Didn't we introduce a timing measuring API to store this kind of data in an easily parsable way?

Reply:
You are right that this does not use system resolver, there is a fallback to localhost tho. I will correct the string, also I am willing to change this to debug if you think it's really not something that should appear every DNS creation.

I am not aware of any kind of timing API, if you mean telemetry API that has been added only to Foreman core. Of course, once we add it here (and it's on my long-term todo) this will be included.